### PR TITLE
Add Service Catalog Support

### PR DESCRIPTION
**Pull request by tomvachon**

### CODE REVIEW

File: resources/servicecatalog-portfolio-share-attachements.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		resp, err := svc.ListPortfolioAccess(accessParams)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, accountID := range resp.AccountIds {
+			accounts = append(accounts, accountID)
+		}
+
+		for _, account := range accounts {
```
Comment (bethge) [inline resources/servicecatalog-portfolio-share-attachements.go]: I would suggest dropping the `accounts` slice and directly iterating over `resp.AccountIds`.

**Comment by tomvachon** [comment]
Updated @bethge, good call.

**Comment by svenwltr** [comment]
I get these errors:

```
ERRO[0013] Listing ServiceCatalogTagOption failed. Please report this to https://github.com/rebuy-de/aws-nuke/issues/new.
    !!! TagOptionNotMigratedException: TagOption Migration not complete
    !!! 	status code: 400, request id: 7ed6b28e-2db5-11e8-b2d9-a55c524cba84 
ERRO[0013] Listing ServiceCatalogTagOptionPortfolioAttachment failed. Please report this to https://github.com/rebuy-de/aws-nuke/issues/new.
    !!! TagOptionNotMigratedException: TagOption Migration not complete
    !!! 	status code: 400, request id: 7efdc28f-2db5-11e8-b2d9-a55c524cba84 
ERRO[0015] Listing ServiceCatalogTagOption failed. Please report this to https://github.com/rebuy-de/aws-nuke/issues/new.
    !!! TagOptionNotMigratedException: TagOption Migration not complete
    !!! 	status code: 400, request id: 7fc92899-2db5-11e8-a112-ef8c097b07ec 
ERRO[0015] Listing ServiceCatalogTagOptionPortfolioAttachment failed. Please report this to https://github.com/rebuy-de/aws-nuke/issues/new.
    !!! TagOptionNotMigratedException: TagOption Migration not complete
    !!! 	status code: 400, request id: 80165dca-2db5-11e8-b012-070f02466997 
```

Verbose output for `ServiceCatalogTagOption`:

```
DEBU[0005] sending AWS request:
    > POST / HTTP/1.1
    > Host: servicecatalog.eu-west-1.amazonaws.com
    > Authorization: <hidden>
    > Content-Length: 15
    > Content-Type: application/x-amz-json-1.1
    > User-Agent: aws-sdk-go/1.13.18 (go1.10; linux; amd64)
    > X-Amz-Date: 20180322T094431Z
    > X-Amz-Target: AWS242ServiceCatalogService.ListTagOptions
    > 
    > {"PageSize":20} 
DEBU[0005] received AWS response:
    < HTTP/1.1 400 Bad Request
    < Connection: close
    < Content-Length: 87
    < Content-Type: application/x-amz-json-1.1
    < Date: Thu, 22 Mar 2018 09:44:31 GMT
    < X-Amzn-Requestid: 9f058668-2db5-11e8-bd43-d9df054031e2
    < 
    < {"__type":"TagOptionNotMigratedException","Message":"TagOption Migration not complete"} 
ERRO[0005] Listing ServiceCatalogTagOption failed. Please report this to https://github.com/rebuy-de/aws-nuke/issues/new.
    !!! TagOptionNotMigratedException: TagOption Migration not complete
    !!! 	status code: 400, request id: 9f058668-2db5-11e8-bd43-d9df054031e2 
DEBU[0005] sending AWS request:
    > POST / HTTP/1.1
    > Host: servicecatalog.us-east-1.amazonaws.com
    > Authorization: <hidden>
    > Content-Length: 15
    > Content-Type: application/x-amz-json-1.1
    > User-Agent: aws-sdk-go/1.13.18 (go1.10; linux; amd64)
    > X-Amz-Date: 20180322T094431Z
    > X-Amz-Target: AWS242ServiceCatalogService.ListTagOptions
    > 
    > {"PageSize":20} 
DEBU[0005] received AWS response:
    < HTTP/1.1 400 Bad Request
    < Connection: close
    < Content-Length: 87
    < Content-Type: application/x-amz-json-1.1
    < Date: Thu, 22 Mar 2018 09:44:32 GMT
    < X-Amzn-Requestid: 9f4f878d-2db5-11e8-9c14-e1445de5da74
    < 
    < {"__type":"TagOptionNotMigratedException","Message":"TagOption Migration not complete"} 
ERRO[0005] Listing ServiceCatalogTagOption failed. Please report this to https://github.com/rebuy-de/aws-nuke/issues/new.
    !!! TagOptionNotMigratedException: TagOption Migration not complete
    !!! 	status code: 400, request id: 9f4f878d-2db5-11e8-9c14-e1445de5da74 
```

**Comment by tomvachon** [comment]
Huh. I’ll have to research that error. Seems like that’s more internal. How
old is this account? Sc went through tons of changes in the past 6 months

**Comment by tomvachon** [comment]
@svenwltr not sure I can solve this one: 
```
An operation requiring TagOptions failed because the TagOptions migration process has not been performed for this account. Please use the AWS console to perform the migration process before retrying the operation.
```

That is a legacy breaking change by AWS, so you have to migrate before you can delete them.  Sometime likely in July 2017 is when that change was introduced.  I think we have to accept this one as a wont-fix

**Comment by svenwltr** [comment]
@tomvachon The account is almost a year old. The thing is, that I never even used the Service Catalog, so  this is kind of confusing. Can we filter the resources somehow if it is an old account?

**Comment by tomvachon** [comment]
@svenwltr I'm not sure how.  Every single call throws that error and no call will respond without that error if the condition is hit.  Do you have a proposal?

**Comment by svenwltr** [comment]
Ok, if it not possible to detect this during the list, we should at least catch this error (in the `Remove()` function) and print a warning instead.

I can do this if you want.

**Comment by tomvachon** [comment]
@svenwltr if you can do that, great.  The problem is you get this on List too, not just delete don't you?

**Comment by svenwltr** [comment]
@tomvachon I added a proper error handling. Actually it was only required in the list. Also I added a log output, but I am not sure if we need that.

/cc @rebuy-de/prp-aws-nuke

**Pull request merged**